blueocean-dashboard/src/main/js/creation/CreatePipelineScmListRenderer.jsx

```diff
@@ -37,8 +51,18 @@ export class CreatePipelineScmListRenderer extends React.Component {
 
             providers = providers.filter(provider => !!provider);
 
+            // eslint-disable-next-line
+            let sortedProviders = [];
+
+            this.customSortProviders(providers, sortedProviders, 'BbCloudScmProvider');
+            this.customSortProviders(providers, sortedProviders, 'BbServerScmProvider');
+            this.customSortProviders(providers, sortedProviders, 'GithubScmProvider');
+            this.customSortProviders(providers, sortedProviders, 'GithubEnterpriseScmProvider');
+            this.customSortProviders(providers, sortedProviders, 'GitScmProvider');
+            this.customSortProviders(providers, sortedProviders); // add all other providers
```

Comment:
I think perhaps we should use the `ordinal` property in `jenkins-js-extension.yaml` as it's the standard way to sort extensions for example in `Extensions.Renderer`. See https://github.com/jenkinsci/blueocean-plugin/blob/master/blueocean-personalization/src/main/js/jenkins-js-extension.yaml for an example how to specify it. Then we should just be able to add a `providers.sort` call below the `providers.filter` call.